api/src/test/java/org/pmiops/workbench/survey/NewUserSatisfactionSurveyServiceTest.java

```diff
@@ -0,0 +1,100 @@
+package org.pmiops.workbench.survey;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.sql.Timestamp;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.pmiops.workbench.access.AccessTierService;
+import org.pmiops.workbench.db.dao.AccessTierDao;
+import org.pmiops.workbench.db.dao.UserAccessTierDao;
+import org.pmiops.workbench.db.model.DbAccessTier;
+import org.pmiops.workbench.db.model.DbNewUserSatisfactionSurvey;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbUserAccessTier;
+import org.pmiops.workbench.test.FakeClock;
+import org.pmiops.workbench.utils.TestMockFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+@SpringJUnitConfig
+class NewUserSatisfactionSurveyServiceTest {
+  @Autowired private NewUserSatisfactionSurveyService newUserSatisfactionSurveyService;
+  @MockBean private AccessTierDao mockAccessTierDao;
+  @MockBean private UserAccessTierDao mockUserAccessTierDao;
+  @MockBean private AccessTierService mockAccessTierService;
+  private DbAccessTier registeredAccessTier;
+
+  private static final Instant START_INSTANT = Instant.parse("2000-01-01T00:00:00.00Z");
+  private static final FakeClock PROVIDED_CLOCK = new FakeClock(START_INSTANT);
+  private DbUser user;
+
+  @TestConfiguration
+  @Import({NewUserSatisfactionSurveyServiceImpl.class})
+  public static class Config {
+    @Bean
+    Clock clock() {
+      return PROVIDED_CLOCK;
+    }
+  }
+
+  @BeforeEach
+  public void setup() {
+    this.registeredAccessTier = TestMockFactory.createRegisteredTierForTests(mockAccessTierDao);
+    when(mockAccessTierService.getRegisteredTierOrThrow()).thenReturn(registeredAccessTier);
+    PROVIDED_CLOCK.setInstant(START_INSTANT);
+    user = new DbUser();
+  }
+
+  @Test
+  public void testEligibleToTakeSurvey_eligible() {
```

Comment:
please add comments here as well describing both the eligibility target window and the particulars of what each test is checking.

Reply:
I'm never sure how to add a "success" case for methods like this. I added this test to verify that yes, under some circumstance, the method returns true. However, the specifics of those values aren't important because they're individually tested by the other (negative) tests.

I added comments to the negative cases. Let me know what I should do with the "eligible" case, if anything.

Reply:
the negative-case comments are great, thanks.

For positive, something like "the enabled date is in the eligibility window" would work